mos-platform/mega65/mega65.h

```diff
@@ -0,0 +1,229 @@
+// Copyright 2023 LLVM-MOS Project
+// Licensed under the Apache License, Version 2.0 with LLVM Exceptions.
+// See https://github.com/llvm-mos/llvm-mos-sdk/blob/main/LICENSE for license
+// information.
+
+/*
+MIT License
+
+Copyright (c) 2023 Mikael Lund aka Wombat
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+#ifndef _MEGA65_H
+#define _MEGA65_H
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wgnu-anonymous-struct"
+#pragma clang diagnostic ignored "-Wnested-anon-types"
+#pragma clang diagnostic ignored "-Wgnu-binary-literal"
+#pragma clang diagnostic ignored "-Wfixed-enum-extension"
+#endif
+
+#include <_45E100.h>
```

Comment:
As I understand it these underscore first files are commonly used in cc65 community. I never liked it and think that it is better to prefix it with a path, like `mega65/6526.h`, compared to `_6526.h`.
To me it seems clear that `mega65/6526.h` is a header that defines the 6526 chip to be used on the MEGA65.
The `_6526.h` leaves me in doubt whether the included header file works with the MEGA65, it may define things for a 6526, but if it provides a base symbol, will it work with the MEGA65, it is not clear.
I do not mind having all these `_6526.h` and whatever if there is a compatibility reason for it, but I would prefer the blessed interface would be "system/chip.h" and it can delegate as needed. 
I honestly do not know if it applies here, but at least it would be nice if the public API was this way.

Reply:
The naming follows the existing style which is inherited from CC65 with the main argument of compatibility. My understanding is that `_` hints "private" and a user would only include `mega65.h` or `c64.h` which is unambiguous. For the specific case of `_6526.h` it resides in the C64 target and thus valid for multiple platforms. It might actually be better to elevate this and the vic chips to `commodore` as they are used by C64, MEGA65, and the currently unimplemented C128. I will leave that for another PR.

Reply:
Ah, I see. Yeah, it kind of works with "private". I suppose the idea is to include `<mega65.h>` or `<c64.h>` if you are on C64. I suppose they could be considered to move in under `commodore` (or perhaps `mos` as the chip manufacturer?) at some point as you suggest.